fe/fe-core/src/main/java/com/starrocks/lake/LakeTableHelper.java

```diff
@@ -157,9 +161,30 @@ static boolean removePartitionDirectory(Partition partition, long warehouseId) t
                 ret = false;
             }
         }
+        if (ret) {
+            // try to remove shard group meta (shards meta included)
+            deleteShardGroupMeta(partition);
```

Comment:
shall not do it here. move it to one-level up who calls the `removePartitionDirectory`

reason: `removePartitionDirectory` doesn't necessary mean to delete related meta, but the upper level deletion of the partition has the explicit meaning of "delete data and delete meta"